src/hotspot/share/gc/g1/g1ConcurrentRefine.hpp

```diff
@@ -43,7 +44,7 @@ class ThreadClosure;
 // iterate over them.
 class G1ConcurrentRefineThreadControl {
   G1ConcurrentRefine* _cr;
-  G1ConcurrentRefineThread** _threads;
+  GrowableArray<G1ConcurrentRefineThread*>* _threads;
```

Comment:
Rather than a pointer to a GA, instead make it a GA value.  And remove _max_num_threads
member, because we can use the GA's capacity for that.

Also, it should be a GrowableArrayCHeap<G1ConcurrentRefineThread*, mtGC>.

I also think the max-threads value should be passed to the constructor, and the thread initialized
with that as it's capacity and a size of zero.

That's going to affect the changes a lot, so I've not bothered even looking at them yet.